packages/api/cms-api/src/builds/changes-checker.console.ts

```diff
@@ -21,9 +26,35 @@ export class ChangesCheckerConsole {
         console.log("Checking if changes since last build occurred...");
 
         if (await this.buildsService.hasChangesSinceLastBuild()) {
-            console.log("Changes detected, starting build...");
-            await this.buildsService.createBuildsForAllScopes("changesDetected");
-            console.log("Build successfully started, resetting changesSinceLastBuild...");
+            if (await this.buildsService.shouldRebuildAllScopes()) {
+                console.log("Starting build(s) for all scopes...");
+                await this.buildsService.createBuildsForAllScopes("changesDetected");
+            } else {
+                const builderCronJobs = await this.buildTemplateService.getAllBuilderCronJobs();
+
+                const getMatchingBuilderCronJob = (scope: ContentScope) => {
+                    for (const cronJob of builderCronJobs) {
+                        const cronJobScope = this.kubernetesService.getContentScope(cronJob);
+
+                        // Checks if scopes match partially. For instance, a job's scope may be { "domain": "main" }, but the change was in
+                        // { "domain": "main", "language": "en" }. Or the job's scope may be { "domain": "main", "language": "en" }, but the change
+                        // was in { "domain": "main" }. In both cases, the job should still be started.
+                        if (Object.entries(cronJobScope).some(([key, value]) => (scope as Record<string, unknown>)[key] === value)) {
```

Comment:
@nsams I need this cast because [ContentScope](https://github.com/vivid-planet/comet/blob/022df773c6523be6e937408e294d90de9399a785/packages/api/cms-api/src/common/decorators/content-scope.interface.ts) is missing an index signature... is it safe to add it?

Reply:
yes, we could even restrict values (for this check to work correctly - exclude nested objects/arrays)

Reply:
Okay, I'll try this in a follow up PR